server/api/emailLists.js

```diff
@@ -148,6 +150,52 @@ const getHeldMessages = listName => (
   })
 );
 
+const getHeldMessageContent = (listName, messageId) => (
+  // Content returned is limited to a prefix of headers and body
+  // with lines possibly truncated, to give enough text for message review
+  // for moderation while preventing load of large messages (such as with
+  // very large file attachments).
+  new Promise((resolve) => {
+    execFile('dumpdb', [MAILMAN_DATA_DIR + '/' + `heldmsg-${listName}-${messageId}.pck`], (error, stdout, stderr) => {
+      if (error) {
+        console.error('!!! get held content error', error, stderr);
+        return resolve([]);
```

Comment:
Any reason to not use a `reject` here?

Reply:
I think not. I'm afraid I was just following the pattern in the function I based the new one on, which I now see is inconsistent with other places in the file and not appropriate for this situation. Where would a catch go exactly if this may reject?

Reply:
I figured out that there is already a catch set up by register() so all set.